lib/libfrr.c

```diff
@@ -1125,8 +1125,8 @@ static void frr_terminal_close(int isexit)
 		 * don't redirect when stdout is set with --log stdout
 		 */
 		for (fd = 2; fd >= 0; fd--)
-			if (isatty(fd) &&
-			    (fd != STDOUT_FILENO || !logging_to_stdout))
+			if (!logging_to_stdout || !isatty(fd) ||
```

Comment:
With this new logic, we redirect stderr, and stdin regardless of whether this is a terminal. Is that a correct behavior?

Reply:
I'm confused.

>As a reminder, before clang-format invocation it was...

clang-formatter should not touch the logic, only formatting.

Reply:
Yes. What I'm talking about, is that clang-format inverted logic and it became a bit more difficult to read. IMO.

Reply:
clang-format does not invert logic, as @ton31337 stated clang-format just moves some stuff around to make it look prettier based upon some basic rules

Reply:
It turned this:
```c
if (logging_to_stdout && isatty(fd) && fd == STDOUT_FILENO) {
    /* Do nothing. */
} else {
    dup2(nullfd, fd);
}
```
into this:
```c
if (!logging_to_stdout || !isatty(fd) || fd != STDOUT_FILENO) {
    dup2(nullfd, fd);
}
```
This is why I called it "invert". Maybe it's not an accurate term.
In any case the final code behavior is the same.

Reply:
I tried to run this code:
```c
if (logging_to_stdout && isatty(fd) && fd == STDOUT_FILENO) {
    /* Do nothing. */
} else {
    dup2(nullfd, fd);
}
```
via clang-formatter, and I don't see anything similar you had (changing the code by clang-formatter).

You can just apply the formatted that is suggested by our frrbot (Github check).

Reply:
The current version is final and all checks are green.

Reply:
@ton31337, should I do something else to go ahead with this PR?

Reply:
I'd like to have a first code instead of the current:

```c
if (logging_to_stdout && isatty(fd) && fd == STDOUT_FILENO) {
    /* Do nothing. */
} else {
    dup2(nullfd, fd);
}
```

Reply:
@ton31337, I've pushed the style you requested, but "Verify Source" checker complains:
```
 Report for libfrr.c | 4 issues
===============================================
< WARNING: braces {} are not necessary for any arm of this statement
< #1128: FILE: /tmp/f1-1772815/libfrr.c:1128:
< WARNING: braces {} are not necessary for any arm of this statement
< #1218: FILE: /tmp/f1-1772815/libfrr.c:1218:
Checkpatch found Errors:
Report for libfrr.c | 4 issues
===============================================
< WARNING: braces {} are not necessary for any arm of this statement
< #1128: FILE: /tmp/f1-1772815/libfrr.c:1128:
< WARNING: braces {} are not necessary for any arm of this statement
< #1218: FILE: /tmp/f1-1772815/libfrr.c:1218:
```

Reply:
Now, done.